src/kademlia/dht_storage.cpp

```diff
@@ -453,6 +464,15 @@ namespace
 			touch_item(i->second, addr);
 		}
 
+		int get_infohashes_sample(entry& item) const override
```

Comment:
I don't really like ``entry`` all that much. Do you think it would be inconvenient to return these values in a more "decoded" form, and defer decoding to the caller?